### ISSUE TITLE
test

### ISSUE THREAD

**Issue by kimho1wq** [kimho1wq/TIL#2]
# EBM
This repository contains official pytorch implementation for following paper:
  - Title : Enhancing Audio Deepfake Detection by Improving Representation Similarity of Bonafide Speech
  - Autor : Seung-bin Kim, Hyun-seo Shin, Jungwoo Heo, Chan-yeong Lim, Kyo-Won Koo, Jisoo Son, Sanghyun Hong, Souhwan Jung, and Ha-Jin Yu

# Abstract
![Image](https://github.com/user-attachments/assets/4de70d04-edc3-4290-8521-5d52c0a00f1c)

The key to audio deepfake detection is distinguishing bonafide speech from carefully generated spoofed speech.
The more distinguishable they are, the better and more generalizable the detection becomes.
In this work, we propose a novel approach to enhance this distinguishability in the latent space.
Inspired by one-class classification, we formulate an objective function that encourages the contraction of bonafide samples while dispersing fake speech samples during training.
Our objective consists of two key components: Bonafide-Pair Learning (BPL) loss and an Extended One-Class Softmax (EOC-S) loss.
The BPL reduces intra-class variance by aligning the embeddings of augmented bonafide pairs, while the EOC-S leverages Adam-based centroid updates and margin constraints to reinforce separability from spoofed data.
Experimental results on ASVspoof datasets demonstrate that our proposed approach enhances detection performance across diverse attack scenarios.

Our experimental code was modified based on [here](https://github.com/talkingnow/HM-Conformer).

# Data
The [ASVspoof 2019](http://www.robots.ox.ac.uk/~vgg/data/voxceleb/) datasets were used for training and test.

The train list should contain the identity and the file path, one line per utterance, as follows:
```
id00000 id00000/youtube_key/12345.wav
id00012 id00012/21Uxsk56VDQ/00001.wav
```
The train list for VoxCeleb2 can be download from [here](http://www.robots.ox.ac.uk/~vgg/data/voxceleb/meta/train_list.txt), and the test lists for VoxCeleb1 can be downloaded from [here](https://mm.kaist.ac.kr/datasets/voxceleb/index.html#testlist). 

For data augmentation, the following script can be used to download and prepare.
```
python3 ./dataprep.py --save_path data --augment
```

We also performed an out-of-domain evaluation using the [VOiCES](https://iqtlabs.github.io/voices/downloads/) development set.

Each dataset must be downloaded in advance for training and testing, and its path must be mapped to the docker environment.

# Environment
Docker image (nvcr.io/nvidia/pytorch:23.07-py3) of Nvidia GPU Cloud was used for conducting our experiments.

Make docker image and activate docker container.
```
./docker/build.sh
./docker/run.sh
```

Note that you need to modify the mapping path before running the 'run.sh' file.

# Training

- MR-RawNet on a single GPU
```
python3 ./trainSpeakerNet.py --config ./configs/MR_RawNet.yaml
```

- MR-RawNet on multiple GPUs
```
CUDA_VISIBLE_DEVICES=0,1 python3 ./trainSpeakerNet.py --config ./configs/MR_RawNet.yaml --distributed
```
Use --distributed flag to enable distributed training.
If you are running more than one distributed training session, you need to change the --port argument.

Note that the configuration file overrides the arguments passed via command line.

# Test

The following script should return: `EER 0.8294`.
```
python3 ./trainSpeakerNet.py --eval --config ./configs/MR_RawNet.yaml --initial_model MR_RawNet.pt
```

# Citation
Please cite if you make use of the code.

```
@article{kim2025ebm,
  title={MR-RawNet: Speaker verification system with multiple temporal resolutions for variable duration utterances using raw waveforms},
  author={Kim, Seung-bin and Lim, Chan-yeong and Heo, Jungwoo and Kim, Ju-ho and Shin, Hyun-seo and Koo, Kyo-Won and Yu, Ha-Jin},
  journal={arXiv preprint arXiv:2406.07103},
  year={2024}
}

```
